$OpenBSD: patch-tests_framework_c,v 1.3 2017/06/21 11:28:06 espie Exp $

XXX needs SIOCGIFINDEX
- Add missing headers.
- Set interface index to -1 by default.

Index: tests/framework.c
--- tests/framework.c.orig
+++ tests/framework.c
@@ -24,8 +24,13 @@
 #include <avahi-client/client.h>
 #include <avahi-common/error.h>
 
-#include <net/if.h>
+#include <sys/types.h>
+#include <sys/socket.h>
+#include <sys/sockio.h>
 #include <sys/ioctl.h>
+#include <net/if.h>
+#include <net/if_dl.h>
+#include <ifaddrs.h>
 
 #include <errno.h>
 #include <unistd.h>
@@ -188,12 +193,14 @@ epc_test_list_ifaces (void)
           continue;
         }
 
+#ifdef SIOCGIFINDEX
       if (ioctl (fd, SIOCGIFINDEX, req) < 0)
         {
           g_warning ("%s: ioctl(SIOCGIFINDEX): %s",
                      G_STRLOC, g_strerror (errno));
           goto out;
         }
+#endif
 
 /* The following allows to fall back to ifr_index on kFreeBSD, where 
    ifr_ifindex does not exist. See bug #592474. */
@@ -201,7 +208,7 @@ epc_test_list_ifaces (void)
 #define ifr_ifindex ifr_index
 #endif
 
-      ifaces[j].ifidx = req->ifr_ifindex;
+      ifaces[j].ifidx = -1;
       ifaces[j].mask = epc_test_result;
 
       g_print ("%s: name=%s, ifidx=%u, \n",
